### PR TITLE
allow users to set topology environment properties

### PR DESCRIPTION
**Pull request by jerrypeng**
I added TOPOLOGY_ENVIRONMENT to make heron more compatible with Storm since Storm has this feature.

We can use TOPOLOGY_WORKER_CHILDOPTS to set these environment properties but it might not be an elegant way of doing it. We would essentially have to put all the user defined environment properties on the java command line that starts the heron instance in a format: -Denv.variable=foo . If the user has many environment properties, this could cause the java command line to start the heron instance to be very large. I don't know if that is a good idea. Doing a System.setProperty in the start of the heron instance seems like a more eloquent approach

### CODE REVIEW

File: heron/api/src/java/com/twitter/heron/api/Config.java [hunk 1]
```diff
@@ -235,6 +235,13 @@
   public static final String TOPOLOGY_UPDATE_REACTIVATE_WAIT_SECS =
       "topology.update.reactivate.wait.secs";
 
+  /**
+   * Topology-specific environment variables for the worker child process.
```
Comment (kramasamy) [inline heron/api/src/java/com/twitter/heron/api/Config.java]: There is no notion of worker child process in heron - can you rephrase it?

File: storm-compatibility/src/java/org/apache/storm/Config.java [hunk 2]
```diff
@@ -357,6 +357,12 @@
       = "topology.bolts.watermark.event.interval.ms";
 
   /**
+   * Topology-specific environment variables for the worker child process.
```
Comment (kramasamy) [inline storm-compatibility/src/java/org/apache/storm/Config.java]: the same as before

**Comment by maosongfu** [comment]
Quick questions: 
Is it feasible to pass env variables by using `TOPOLOGY_WORKER_CHILDOPTS ` and `TOPOLOGY_COMPONENT_JVMOPTS` in Config: https://github.com/twitter/heron/blob/master/heron/api/src/java/com/twitter/heron/api/Config.java#L51?

What are the major benefits to have a specific Config to allow users to set topology environment properties? (Can you also put this in the description of this pull request)

File: storm-compatibility/src/java/org/apache/storm/Config.java (hunk 2 above)
Comment (jerrypeng) [reply storm-compatibility/src/java/org/apache/storm/Config.java]: Lets keep this same as it is for Storm

File: heron/api/src/java/com/twitter/heron/api/Config.java (hunk 1 above)
Comment (jerrypeng) [reply heron/api/src/java/com/twitter/heron/api/Config.java]: will fix

**Comment by jerrypeng** [comment]
@maosongfu I added TOPOLOGY_ENVIRONMENT to make heron more compatible with Storm since Storm has this feature. 

We can use TOPOLOGY_WORKER_CHILDOPTS to set these environment properties but it might not be an elegant way of doing it.  We would essentially have to put all the user defined environment properties on the java command line that starts the heron instance in a format:  -Denv.variable=foo .  If the user has many environment properties, this could cause the java command line to start the heron instance to be very large.  I don't know if that is a good idea.

**Comment by maosongfu** [comment]
@jerrypeng sounds fair enough. Looks good to me. Can you also add the justification you mentioned in the description of this pull request?

**Comment by jerrypeng** [comment]
@maosongfu sounds good!  I have added the description.  Thanks for reviewing!
